sdk/tests/core/test_az_json.c

```diff
@@ -3049,15 +3061,37 @@ static void test_az_json_string_unescape(void** state)
     destination_span._internal.size = final_size;
 
     assert_int_equal(AZ_OK, result);
-    assert_int_equal(expected._internal.size, final_size);
-    assert_string_equal(expected._internal.ptr, destination);
     assert_true(az_span_is_content_equal(expected, destination_span));
   }
 
   // fake escapes
   {
     az_span original = AZ_SPAN_FROM_STR("\\9");
-    az_span expected = AZ_SPAN_FROM_STR("\\9");
+
+    char destination[59] = { 0 };
+
+    int final_size;
+    az_result result = az_json_string_unescape(original, destination, 59, &final_size);
+
+    assert_int_equal(AZ_ERROR_UNEXPECTED_CHAR, result);
+  }
+
+  // malformed escapes
+  {
+    az_span original = AZ_SPAN_FROM_STR("abcd\\");
+
+    char destination[59] = { 0 };
+
+    int final_size;
+    az_result result = az_json_string_unescape(original, destination, 59, &final_size);
+
+    assert_int_equal(AZ_ERROR_UNEXPECTED_END, result);
+  }
+
+  // malformed escapes 2
```

Comment:
This comment doesn't match the test case ("a")

Reply:
done